pkg/cmd/root.go

```diff
@@ -27,6 +29,11 @@ func NewCmd() *cobra.Command {
 		Short:                  "ike lets you safely develop and test on prod without a sweat",
 		BashCompletionFunction: completion.BashCompletionFunc,
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error { //nolint[:unparam]
+			if v.Released() && !version.IsLatestRelease() {
```

Comment:
Calling GitHub twice, IsLatestRelease and LatestRelease

Maybe instead:

latest := LatestRelease()
if version.IsLatest(latest) {}

Reply:
On missing network, this would potentially timeout twice

Reply:
This being in the root command.. that would force 'serve' to check for version on startup too?

Reply:
Could we do it 'async' in Pre and show the message in Post or something to avoid it interfering with normal operations?

Reply:
> This being in the root command.. that would force 'serve' to check for version on startup too?

Yup, is that a concern? It won't fail, will log in case of network issues.

Reply:
> Could we do it 'async' in Pre and show the message in Post or something to avoid it interfering with normal operations?

I will have a look. Thanks for the hint.

Reply:
> Maybe instead:
> 
> latest := LatestRelease()
> if version.IsLatest(latest) {}

Good point.

Reply:
Single gh call fixed in 694dfd8

Reply:
> > Could we do it 'async' in Pre and show the message in Post or something to avoid it interfering with normal operations?
> 
> I will have a look. Thanks for the hint.

Or maybe even simpler have a flag to turn it off. On by default but can be turned off.

Reply:
> Yup, is that a concern? It won't fail, will log in case of network issues.

The concern is more around it wouldn't be the first 'un-required' thing in a program to fail or slow it down to a dog. Good intentions and all but.. Let's ensure it doesn't get in the way.

Reply:
Reworked in 9369c11